### ISSUE TITLE
BUG: Hotkey does nothing, apparently after earlier exception

### ISSUE THREAD

**Issue by Kylotan** [Flow-Launcher/Flow.Launcher#2223]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I am using the latest version of Flow Launcher.

### Problem Description

Sometimes when I use the hotkey (Alt+Space in my case) to bring up the Flow Launcher search bar, it doesn't respond. Flow Launcher is still running, appearing in Task Manager and in the system notification area, but it does not respond to the hotkey.

When the process is killed and restarted, it works fine again.

### To Reproduce

Reproduction steps are unknown. However, when I see this behavior, inspecting the most recent log shows exceptions, some time earlier. (i.e. These errors are not caused by using the hotkey.) These appear in batches every 10 minutes, perhaps linked to periodic indexing operations?

### Flow Launcher Version

1.15.0

### Windows Build Number

10.0.19045.3086

### Error Log

<details>

```shell

[these exceptions appear quite soon after startup, and I believe the program is still working properly after these.]

20:52:02.0986+01:00 - ERROR -  - ------------- BEGIN Flow.Launcher.Plugin.Program exception -------------
20:52:02.0986+01:00 - ERROR -  - 
Exception full name: System.Runtime.InteropServices.COMException
Error status: UNKNOWN
Class name: IShellLinkW
Calling method: retrieveTargetPath
Program path: C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Accessories\Windows Media Player.lnk
InnerException number: 1
Exception message: Error HRESULT E_FAIL has been returned from a call to a COM component.
Exception error type: HResult -2147467259
Exception thrown in called method: Void GetDescription(System.Text.StringBuilder, Int32)
Possible interpretation of the error: Error caused likely due to trying to get the description of the program
Possible resolution: Not yet known
20:52:02.0986+01:00 - ERROR -  - ------------- END Flow.Launcher.Plugin.Program exception -------------

20:52:02.2278+01:00 - ERROR -  - ------------- BEGIN Flow.Launcher.Plugin.Program exception -------------
20:52:02.2278+01:00 - ERROR -  - 
Exception full name: System.Runtime.InteropServices.COMException
Error status: UNKNOWN
Class name: IShellLinkW
Calling method: retrieveTargetPath
Program path: C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Administrative Tools\ODBC Data Sources (32-bit).lnk
InnerException number: 1
Exception message: Error HRESULT E_FAIL has been returned from a call to a COM component.
Exception error type: HResult -2147467259
Exception thrown in called method: Void GetDescription(System.Text.StringBuilder, Int32)
Possible interpretation of the error: Error caused likely due to trying to get the description of the program
Possible resolution: Not yet known
20:52:02.2278+01:00 - ERROR -  - ------------- END Flow.Launcher.Plugin.Program exception -------------

[...........followed by many of these exceptions later...., and I think it's after one or more of these that the search bar stops appearing]

23:14:36.1035+01:00 - ERROR - ImageLoader.Load - Failed to get thumbnail for C:\Program Files (x86)\Steam\steam\games\b0ff6bdb2da01a5924a7f67359af45b8c69ff5b9.ico on second try
EXCEPTION OCCURS: System.Runtime.InteropServices.COMException (0x80004005): Error while extracting thumbnail for C:\Program Files (x86)\Steam\steam\games\b0ff6bdb2da01a5924a7f67359af45b8c69ff5b9.ico
 ---> System.Runtime.InteropServices.COMException (0x8004B201): 0x8004B201
   at IntPtr Flow.Launcher.Infrastructure.Image.WindowsThumbnailProvider.GetHBitmap(string fileName, int width, int height, ThumbnailOptions options) in C:/projects/flow-launcher/Flow.Launcher.Infrastructure/Image/ThumbnailReader.cs:line 129
   at BitmapSource Flow.Launcher.Infrastructure.Image.WindowsThumbnailProvider.GetThumbnail(string fileName, int width, int height, ThumbnailOptions options) in C:/projects/flow-launcher/Flow.Launcher.Infrastructure/Image/ThumbnailReader.cs:line 120
   at ImageResult Flow.Launcher.Infrastructure.Image.ImageLoader.GetThumbnailResult(ref string path, bool loadFullImage) in C:/projects/flow-launcher/Flow.Launcher.Infrastructure/Image/ImageLoader.cs:line 239
   at async ValueTask<ImageResult> Flow.Launcher.Infrastructure.Image.ImageLoader.LoadInternalAsync(string path, bool loadFullImage) in C:/projects/flow-launcher/Flow.Launcher.Infrastructure/Image/ImageLoader.cs:line 163

23:24:36.1167+01:00 - ERROR - ImageLoader.Load - Failed to get thumbnail for C:\Program Files (x86)\Steam\steam\games\dca530aba423efd0d3cf5794fe63675e88bf60a0.ico on first try
EXCEPTION OCCURS: System.Runtime.InteropServices.COMException (0x80004005): Error while extracting thumbnail for C:\Program Files (x86)\Steam\steam\games\dca530aba423efd0d3cf5794fe63675e88bf60a0.ico
 ---> System.Runtime.InteropServices.COMException (0x8004B201): 0x8004B201
   at IntPtr Flow.Launcher.Infrastructure.Image.WindowsThumbnailProvider.GetHBitmap(string fileName, int width, int height, ThumbnailOptions options) in C:/projects/flow-launcher/Flow.Launcher.Infrastructure/Image/ThumbnailReader.cs:line 129
   at BitmapSource Flow.Launcher.Infrastructure.Image.WindowsThumbnailProvider.GetThumbnail(string fileName, int width, int height, ThumbnailOptions options) in C:/projects/flow-launcher/Flow.Launcher.Infrastructure/Image/ThumbnailReader.cs:line 120
   at ImageResult Flow.Launcher.Infrastructure.Image.ImageLoader.GetThumbnailResult(ref string path, bool loadFullImage) in C:/projects/flow-launcher/Flow.Launcher.Infrastructure/Image/ImageLoader.cs:line 239
   at async ValueTask<ImageResult> Flow.Launcher.Infrastructure.Image.ImageLoader.LoadInternalAsync(string path, bool loadFullImage) in C:/projects/flow-launcher/Flow.Launcher.Infrastructure/Image/ImageLoader.cs:line 163

[...and many other similar exceptions
```
</details>

**Comment by taooceros** [comment]
The log doesn't have any meaningful information unfortunately (they are all expected to happen on some computers and shall not affect how flow works). I think I have seen this bug sometimes, but don't have a clear understanding about why it happens. Maybe the hotkey registration somehow not works after some event.

**Comment by Kylotan** [comment]
I notice that the log doesn't seem to record when the search window is invoked. Might this be a useful addition to the logging? This would allow us to see whether it's a problem with the launching process (e.g. the hotkey) or a problem with the window visibility (e.g. it's offscreen somewhere)

**Comment by deefrawley** [comment]
You don't have Powertoys or any other launcher or comment bound to Alt+Space?

**Comment by Kylotan** [comment]
No. The hotkey works most of the time. This is a sporadic problem.

**Comment by deefrawley** [comment]
Is the issue  still there in the latest version (1.16)?

**Comment by Kylotan** [comment]
I had this happen again, and this was immediately after exiting a session of The Witcher 3. I launched the game via Flow Launcher so it worked okay before the game and stopped working during or immediately afterwards. Killing the Flow Launcher process and restarting it has restored all functionality.

**Comment by jjw24** [comment]
Can you repro it by entering and exit the game?

**Comment by Kylotan** [comment]
No, it works fine if I quit straight back out. It might be something to do with the length of time spent in the game?

**Comment by jjw24** [comment]
If possible perhaps observe next time when you do a longer session. We need to have a way to reliable replicate this in order to troubleshoot efficiently.

**Comment by Kylotan** [comment]
Not sure what there is to observe. I did a longer session but didn't see the same effect this time.

**Comment by jjw24** [comment]
That's ok, let's keep this open for a while and if you could update when you see a pattern to repro this issue.

**Comment by Kylotan** [comment]
Saw this again today, and again it was while I had Witcher 3 open. Definitely a link somehow. This time there was an additional symptom - Flow Launcher opened the search window by itself and froze in a semi-opaque state, and I had to kill it. I relaunched it while Witcher 3 was still open and now it's unresponsive again.

**Comment by VictoriousRaptor** [comment]
> Saw this again today, and again it was while I had Witcher 3 open. Definitely a link somehow. This time there was an additional symptom - Flow Launcher opened the search window by itself and froze in a semi-opaque state, and I had to kill it. I relaunched it while Witcher 3 was still open and now it's unresponsive again.

Do you press alt + space in Witcher 3? And please check if you are using the latest version of Flow (1.17.0).

**Comment by Kylotan** [comment]
No, I don't press they key combination in the game.

The situation I mentioned in the last comment where Flow Launcher launched itself actually happened when I wasn't even touching any input device. The Witcher launcher was just a background window and I was reading something in Firefox at the time.

I've just updated now from 1.16.0 to 1.16.2 but I don't see anything in the release notes that suggests this would have been fixed in those last 2 updates.
